pkg/util/kubernetes/apiserver/common/feature_gate.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package common
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/retry"
+	"k8s.io/client-go/discovery"
+)
+
+const (
+	featureGatesCacheKey = "featureGates"
+
+	// API Server metrics path
+	apiServerMetricsPath = "/metrics"
+)
+
+var (
+	retrier retry.Retrier
+)
+
+// FeatureGate represents a single Kubernetes feature gate
+type FeatureGate struct {
+	Name    string
+	Stage   string
+	Enabled bool
+}
+
+// parseFeatureGatesFromMetrics parses the metrics endpoint, extracting feature gate information
+// Expected format: kubernetes_feature_enabled{name="FeatureName",stage="STAGE"} 1
+func parseFeatureGatesFromMetrics(metricsData []byte) (map[string]FeatureGate, error) {
```

Comment:
like this public functin should do the job for you right ? 
https://github.com/datadog/datadog-agent/blame/5c0ec837482e3536f8fb232ac27efd6aa32866c6/pkg/util/prometheus/parse.go#L51 

it's more complexe to reach the label value

here is an example: https://github.com/datadog/datadog-agent/blame/5c0ec837482e3536f8fb232ac27efd6aa32866c6/pkg/collector/corechecks/containers/containerd/check.go#L206

but it will be structured and upgrade proof for the future 😉